### PR TITLE
#23 Allow duplicate ids during import

### PR DESCRIPTION
**Pull request by kaklakariada**
See #23

### CODE REVIEW

**Comment by kaklakariada** [comment]
Retest this please

**Comment by kaklakariada** [comment]
@redcatbear Please review only the last four commits. The previous commits are already merged into master.

File: src/test/java/openfasttrack/exporter/specobject/TestSpecobjectExporter.java [hunk 1]
```diff
@@ -156,23 +157,22 @@ private void assertExportContent(final String expectedContent,
         LOG.info(() -> "Expected: " + expectedContent);
         assertEquals(expectedContent, actualContent);
         assertThat(actualContent, equalTo(expectedContent));
-        final Collection<SpecificationItem> actualParsedSpecobjects = parseSpecobjectXml(
-                actualContent);
+        final List<SpecificationItem> actualParsedSpecobjects = parseSpecobjectXml(actualContent);
 
-        final Collection<SpecificationItem> expectedItems = Arrays.stream(expectedLinkedItems)
```
Comment (redcatbear) [inline src/test/java/openfasttrack/exporter/specobject/TestSpecobjectExporter.java]: I personally prefer to see when static methods come from a different module.

File: src/test/java/openfasttrack/importer/TestSpecificationListBuilder.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    public void testDuplicateIdNotIgnored()
+    {
+        this.builder.beginSpecificationItem();
+        this.builder.setId(ITEM_ID);
+        this.builder.endSpecificationItem();
+        this.builder.beginSpecificationItem();
+        this.builder.setId(ITEM_ID);
+        this.builder.endSpecificationItem();
+
+        assertThat(this.builder.getItemCount(), equalTo(2));
```
Comment (redcatbear) [inline src/test/java/openfasttrack/importer/TestSpecificationListBuilder.java]: :+1: Nice and minimal.

File: src/test/java/openfasttrack/importer/specobject/TestSpecobjectImporter.java [hunk 3]
```diff
@@ -84,31 +62,27 @@ public void setup()
     @Test
     public void testSingleSpecObject() throws FileNotFoundException
     {
-        final Map<SpecificationItemId, SpecificationItem> result = runImporter(
-                "single-specobject.xml");
-        assertThat(result.values(), hasSize(1));
-        assertThat(result, hasEntry(this.specItem1.getId(), this.specItem1));
+        final List<SpecificationItem> result = runImporter("single-specobject.xml");
```
Comment (redcatbear) [inline src/test/java/openfasttrack/importer/specobject/TestSpecobjectImporter.java]: Now it is an integration test (accesses resource). I think we should follow the Maven naming rules for ITests.

**Comment by redcatbear** [comment]
Review okay. Created ticket: #26

**Pull request merged**